src/proof/alethe/alethe_post_processor.cpp

```diff
@@ -1162,6 +1162,105 @@ bool AletheProofPostprocessCallback::update(Node res,
              && addAletheStepFromOr(
                  AletheRule::DUPLICATED_LITERALS, res, {vp4}, {}, *cdp);
     }
+    //================================================= Equality rules
+    // The following rules are all translated according to the singleton
+    // pattern.
+    case PfRule::REFL:
+    {
+      return addAletheStep(AletheRule::REFL,
+                           res,
+                           nm->mkNode(kind::SEXPR, d_cl, res),
+                           children,
+                           {},
+                           *cdp);
+    }
+    case PfRule::SYMM:
+    {
+      if (res.getKind() == kind::NOT)
+      {
+        return addAletheStep(AletheRule::NOT_SYMM,
+                             res,
+                             nm->mkNode(kind::SEXPR, d_cl, res),
+                             children,
+                             {},
+                             *cdp);
+      }
+      return addAletheStep(AletheRule::SYMM,
+                           res,
+                           nm->mkNode(kind::SEXPR, d_cl, res),
+                           children,
+                           {},
+                           *cdp);
+    }
+    case PfRule::TRANS:
+    {
+      return addAletheStep(AletheRule::TRANS,
+                           res,
+                           nm->mkNode(kind::SEXPR, d_cl, res),
+                           children,
+                           {},
+                           *cdp);
+    }
+      // ======== Congruence
+      // In the case that the kind of the function symbol ?f is forall or
+      // exists, the cong rule needs to be converted into a bind rule. The first
+      // n children will be refl rules, e.g. (= (v0 Int) (w0 Int)).
```

Comment:
isn't it the case that `v0` must be equal to `w0` in the case of a CONG rule on FORALL/EXISTS?

Reply:
Yes it is, otherwise it cannot be proven by a reflexivity rule. However, I changed this from v0 to w0 last minute because I thought that when this rule is in a context that maps w0 to v0 it might be easier to understand this way. Does this make any sense?

Reply:
If I understand correctly I think what you are referring to would only happen to `PfRule::ALPHA_EQUIV` steps, no? It should be the case that for FORALL/EXISTS nodes the first proof child should always be `REFL` (you could add an assertion).

Reply:
Makes sense, let me change this back.